tsl/src/nodes/decompress_chunk/planner.c

```diff
@@ -313,13 +341,33 @@ decompress_chunk_plan_create(PlannerInfo *root, RelOptInfo *rel, CustomPath *pat
 		/* from create_indexscan_plan() */
 		IndexPath *ipath = castNode(IndexPath, compressed_path);
 		ListCell *lc;
+		List *indexqual = NIL;
+		Plan *indexplan;
 		foreach (lc, clauses)
 		{
 			RestrictInfo *rinfo = lfirst_node(RestrictInfo, lc);
 			if (is_redundant_derived_clause(rinfo, ipath->indexclauses))
 				continue; /* dup or derived from same EquivalenceClass */
 			cscan->scan.plan.qual = lappend(cscan->scan.plan.qual, rinfo->clause);
 		}
+		/* the index plan below this node was already created with the
```

Comment:
suggested comment: joininfo clauses have to contain clauses on both compressed and decompressed attnos, but the index scans can't handle compressed attributes, so remove them from the indexscans here.